packages/machine-extractor/src/utils.ts

```diff
@@ -194,32 +205,33 @@ const propertyKey = unionType<{
  * an object expression
  */
 export const getPropertiesOfObjectExpression = (
-  node: t.ObjectExpression | undefined,
+  path: NodePath<t.ObjectExpression> | null | undefined,
   context: ParserContext
 ) => {
   const propertiesToReturn: {
+    path: NodePath<t.ObjectProperty | t.ObjectMethod>;
     node: t.ObjectProperty | t.ObjectMethod;
     key: string;
     keyNode: t.Node;
+    keyPath: NodePath<t.Node>;
     property: t.ObjectMethod | t.ObjectProperty | t.SpreadElement;
+    propertyPath: NodePath<t.ObjectMethod | t.ObjectProperty | t.SpreadElement>;
   }[] = [];
 
-  node?.properties.forEach((property) => {
-    const propertiesToParse: (
-      | t.ObjectMethod
-      | t.ObjectProperty
-      | t.SpreadElement
-    )[] = [property];
+  path?.get("properties").forEach((property) => {
+    const propertiesToParse: NodePath<
+      t.ObjectMethod | t.ObjectProperty | t.SpreadElement
+    >[] = [property];
```

Comment:
nit: technically this is probably slightly incorrect - it should be a union of node paths and not a node path containing. In practice, this might create assignability problems for "correctly" typed functions:
[TS playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgOQgEwKYAUCGMAWANHDAJ5gYDOcO18AvnAGZQQhwDkAAgEY48YANgHoAxtAwcAUFLIU4AQTgBeFOmx58AHhgA6API8AVhlEwsrCrFJwAPiV0BJTADsYwJsAxQAfDKYAri5mwBAuJFQwABQ4AFxqmLgEOgbGpuaW3mQ+dnAJGsl6zhhuHl6+AJSI9DIwkVEIjLSKFUA)